crates/polars-core/src/chunked_array/temporal/date.rs

```diff
@@ -32,7 +32,15 @@ impl DateChunked {
 
     /// Convert from Date into String with the given format.
     /// See [chrono strftime/strptime](https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html).
-    pub fn to_string(&self, format: &str) -> StringChunked {
+    pub fn to_string(&self, format: &str) -> PolarsResult<StringChunked> {
+        write!(
+            String::new(),
+            "{}",
+            NaiveDate::from_num_days_from_ce_opt(0)
+                .unwrap()
+                .format(format)
+        )
+        .map_err(|_| polars_err!(ComputeError: "cannot format Date with format '{}'", format))?;
```

Comment:
this is what's done in the `Datetime` case - try out the format with a placeholder date, and propagate the error (if any) to the user

Reply:
From having tried this out in a plugin, compiled in release mode, I'm not seeing a too meaningful difference.

For just 1000 rows:

- Including this extra check: min: 0.0002998930000709758
- Without this check: min: 0.00029717966632839915

The variance between runs is much greater than the difference. For larger inputs, it vanishes

Script I used to time it:
```
import numpy as np
import timeit

setup = """
import polars as pl
import polars_xdt as xdt
import numpy as np
rng = np.random.default_rng(1)
size = 1000
n = rng.integers(-100_000, 100_000, size=size)
df = pl.DataFrame({'a': n}).select(pl.col('a').cast(pl.Date))
"""

results = np.array(timeit.Timer(
    stmt="df.select(xdt.format_localized('a', '%Y-%m-%d'))",
    setup=setup,
    )
    .repeat(7, 3)
)/3
print(f"min: {min(results)}")
print(f"max: {max(results)}")
print(f"{np.mean(results)} +/- {np.std(results)/np.sqrt(len(results))}")
```

Diff between the scenarios above:
```diff
-            write!(
-                String::new(),
-                "{}",
-                NaiveDate::from_num_days_from_ce_opt(0)
-                    .unwrap()
-                    .format(format)
-            )
-            .map_err(|_| polars_err!(ComputeError: "cannot format Date with format '{}'", format))?;
+            // write!(
+            //     String::new(),
+            //     "{}",
+            //     NaiveDate::from_num_days_from_ce_opt(0)
+            //         .unwrap()
+            //         .format(format)
+            // )
+            // .map_err(|_| polars_err!(ComputeError: "cannot format Date with format '{}'", format))?;
```

Is this acceptable?